{{cookiecutter.project_name}}/environments/kolla/configuration.yml

```diff
@@ -62,4 +62,21 @@ ceph_public_network: {{cookiecutter.ceph_network_frontend}}
 {%- endif %}
 
 ##########################################################
+# Keystone
+keystone_identity_providers:
+  - name: "keycloak"
+    openstack_domain: "keycloak"
+    protocol: "openid"
+    identifier: "https://keycloak.testbed.osism.xyz/auth/realms/osism"
+    public_name: "Authenticate via Keycloak"
+    attribute_mapping: "mappingId1"
+    metadata_folder: "{{ node_custom_config }}/keystone/federation/oidc/metadata"
```

Comment:
RAW blocks are required here to not parse {{ node_custom_config }}.

Have a look at https://github.com/osism/cfg-cookiecutter/blob/main/%7B%7Bcookiecutter.project_name%7D%7D/environments/manager/configuration.yml#L30-L61 for aample. It's fine to have a single RAW block around all keystone parameters here.

Reply:
Added RAW block :+1: